pkg/controller/daemon/daemon_controller.go

```diff
@@ -1427,6 +1427,7 @@ func NewPod(ds *apps.DaemonSet, nodeName string) *v1.Pod {
 // the predicates include:
 //   - PodFitsHost: checks pod's NodeName against node
 //   - PodMatchNodeSelector: checks pod's NodeSelector and NodeAffinity against node
+//   - PodToleratesNodeTaints: exclude tainted node unless pod has specific toleration
```

Comment:
nit: One suggestion that @bsalamat gave was to change the name of this function to reflect that we are checking fitness of node based on the 3 conditions that we have now. I'd recommend `checkNodeFitness` but I don't have strong preference.